### PR TITLE
highlight cache issue when changing theme

### PR DESCRIPTION
**Pull request by seb-odoo**
To reproduce:
- install a theme on a website
- try to install another theme on the same website

### CODE REVIEW

File: odoo/models.py [hunk 1]
```diff
@@ -4888,7 +4888,10 @@ def with_env(self, env):
 
         :type env: :class:`~odoo.api.Environment`
         """
-        return self._browse(env, self._ids, self._prefetch_ids)
+        res = self._browse(env, self._ids, self._prefetch_ids)
+        if self.env.cache != res.env.cache:
+            _logger.warning("Wrong cache after with_env().", stack_info=True)
```
Comment (seb-odoo) [inline odoo/models.py]: When I add here `res.env.cache = self.env.cache` this fixes the issue.

**Comment by seb-odoo** [comment]
@rco-odoo strange cache problem when changing environment, as reported in https://github.com/odoo/odoo/pull/37577#issuecomment-537079831

**Comment by seb-odoo** [comment]
cc @beledouxdenis could you check this please, maybe you will know more too

**Comment by beledouxdenis** [comment]
Cant do atm

**Comment by qsm-odoo** [comment]
@rco-odoo Any insight ? This is really slowing us down... and I will probably need to switch themes for my workshop... :/

**Comment by seb-odoo** [comment]
Rebased and re-targeted from 12.5 to 13.0 as this is an important issue to fix.

**Comment by rdeodoo** [comment]
> Rebased and re-targeted from 12.5 to 13.0 as this is an important issue to fix.

:+1:

**Comment by seb-odoo** [comment]
Theme issue will be handled in https://github.com/odoo/odoo/pull/38209

**Pull request closed without merging**